[msgpack] Fix various vendor_cxx compatibility problems.

For cpp_config_decl.hpp, vendor_cxx gets a bit too greedy and tries to
span its namespace changes across #ifdef branches. We can avoid that
outcome by adding a redundant #include statement to help guide it.


--- include/msgpack/v2/cpp_config_decl.hpp.orig
+++ include/msgpack/v2/cpp_config_decl.hpp
@@ -50,6 +50,8 @@
 
 #else  // MSGPACK_USE_CPP03
 
+#include <memory>
+
 namespace msgpack {
 /// @cond
 MSGPACK_API_VERSION_NAMESPACE(v2) {
--- include/msgpack/v3/cpp_config_decl.hpp.orig
+++ include/msgpack/v3/cpp_config_decl.hpp
@@ -50,6 +50,8 @@
 
 #else  // MSGPACK_USE_CPP03
 
+#include <memory>
+
 namespace msgpack {
 /// @cond
 MSGPACK_API_VERSION_NAMESPACE(v3) {
